spec.bs

```diff
@@ -347,6 +564,209 @@ Moreover, each {{SharedStorageWorklet}}'s [=global scopes|list of global scopes=
     1. Return the value of [=addModule success=].
   </div>
 
+  ## {{SharedStorageUrlWithMetadata}} and Reporting ## {#reporting}
+
+  {{SharedStorageUrlWithMetadata}} [=dictionaries=] contain a [=string=] representing a [=/URL=] and, optionally, a {{reportingMetadata}} {{/object}}.
+
+  <xmp class='idl'>
+    dictionary SharedStorageUrlWithMetadata {
+      required USVString url;
+      object reportingMetadata;
+    };
+  </xmp>
+
+  If a {{SharedStorageUrlWithMetadata}} [=dictionary=] contains a non-[=map/empty=] {{SharedStorageUrlWithMetadata/reportingMetadata}} {{/object}} in the form of a [=dictionary=] whose [=map/keys=] are {{FenceEvent/eventType}}s and whose [=map/values=] are [=strings=] that parse to valid [=/URLs=], then these {{FenceEvent/eventType}}-[=/URL=] pairs will be [=register reporting metadata|registered=] for later access within any [=fenced frame=] that loads the {{SharedStorageResponse}} resulting from this {{SharedStorageWorklet/selectURL()}} call.
```

Comment:
I just barely updated it to _"...whose [=map/keys=] are {{FenceEvent}}'s {{FenceEvent/eventType}}s and..."_. I think none of the fields are directly related to a {{FencedEvent}} when provided by the user. It will be used to construct a FencedEvent at a later stage. This is nevertheless underspecified, and under section "To <dfn>register reporting metadata</dfn>", I've linked to issue https://github.com/WICG/shared-storage/issues/144.